File: frontend/src/admin/departmentLabels.js
/** Department labels — mirrors app/department_labels.py (keep in sync). */

export const DEPARTMENT_LABELS = {
  PTSP: "Phát triển sản phẩm",
  DP: "Điều phối",
  DVA: "Dịch vụ & vận hành",
};

export function departmentLabel(code) {
  if (!code) return "—";
  return DEPARTMENT_LABELS[code] ?? code;
}

export function departmentOptionLabel(code) {
  const label = DEPARTMENT_LABELS[code];
  return label ? `${code} — ${label}` : code;
}

export function departmentDisplay(code) {
  if (!code) return "—";
  const label = DEPARTMENT_LABELS[code];
  return label ? `${code} — ${label}` : code;
}

File: app/department_labels.py
"""Human-readable labels for ``settings.departments`` codes.

Canonical source — keep ``frontend/src/admin/departmentLabels.js`` in sync.
"""

from __future__ import annotations

# code -> short Vietnamese department name
DEPARTMENT_LABELS: dict[str, str] = {
    "PTSP": "Phát triển sản phẩm",
    "DP": "Điều phối",
    "DVA": "Dịch vụ & vận hành",
}


def department_label(code: str | None) -> str:
    if not code:
        return "—"
    return DEPARTMENT_LABELS.get(code, code)


def department_option_label(code: str) -> str:
    """Select option: ``PTSP — Phát triển sản phẩm``."""
    label = DEPARTMENT_LABELS.get(code)
    return f"{code} — {label}" if label else code


def department_display(code: str | None) -> str:
    if not code:
        return "—"
    label = DEPARTMENT_LABELS.get(code)
    return f"{code} — {label}" if label else code
